amt.go

```diff
@@ -458,14 +461,13 @@ func (n *Node) loadNode(ctx context.Context, bs cbor.IpldStore, i uint64, create
 	return subn, nil
 }
 
-func nodesForHeight(width, height int) uint64 {
-	val := math.Pow(float64(width), float64(height))
-	if val >= float64(math.MaxUint64) {
-		log.Errorf("nodesForHeight overflow! This should never happen, please report this if you see this log message")
-		return math.MaxUint64
+func nodesForHeight(height int) uint64 {
+	heightLogTwo := uint64(widthBits * height)
+	if heightLogTwo >= 64 {
+		// Should never happen. Max height is checked at all entry points.
+		panic("height overflow")
```

Comment:
I mean, I think panicing is better than logging and returning the _wrong_ thing, right?

Reply:
probably, yes